### PR TITLE
D2D Scenario Update

### PR DESCRIPTION
**Pull request by dhruvigoyal25**
This PR makes the following changes to AdvantEDGE:
1. Added a new object to Scenario deployment: D2dconfig
2. Support for upgrade scenario

### CODE REVIEW

File: go-packages/meep-model/validator.go [hunk 1]
```diff
@@ -96,6 +98,7 @@ var Version150 = semver.Version{Major: 1, Minor: 5, Patch: 0}
 var Version151 = semver.Version{Major: 1, Minor: 5, Patch: 1}
 var Version153 = semver.Version{Major: 1, Minor: 5, Patch: 3}
 var Version168 = semver.Version{Major: 1, Minor: 6, Patch: 8}
+var Version182 = semver.Version{Major: 1, Minor: 8, Patch: 2}
```
Comment (dilallkx) [inline go-packages/meep-model/validator.go:102]: Current validator version must also be bumped to 1.8.2

File: go-packages/meep-model/validator.go [hunk 2]
```diff
@@ -564,6 +588,16 @@ func validateDeployment(deployment *dataModel.Deployment) (err error) {
 	if err != nil {
 		return errors.New("Invalid connectivity model: " + err.Error())
 	}
+
+	// D2D
+	if deployment.D2d == nil {
+		return errors.New("d2d == nil")
+	} else {
```
Comment (dilallkx) [inline go-packages/meep-model/validator.go:595]: No need for else here.

**Review by dilallkx — CHANGES_REQUESTED** [review]

File: go-packages/meep-model/validator.go (hunk 1 above)
Comment (dhruvigoyal25) [reply go-packages/meep-model/validator.go:102]: Addressed.

**Review by dilallkx — APPROVED** [review]

File: js-apps/meep-frontend/src/js/util/elem-utils.js [hunk 3]
```diff
@@ -250,6 +258,7 @@ export const createElem = name => {
   setElemFieldVal(elem, FIELD_APP_PKT_LOSS, DEFAULT_PACKET_LOSS_APP);
   setElemFieldVal(elem, FIELD_META_DISPLAY_MAP_COLOR, '');
   setElemFieldVal(elem, FIELD_META_DISPLAY_MAP_ICON, '');
+  setElemFieldVal(elem, FIELD_META_DISPLAY_SEQ_PARTICIPANTS, '');
```
Comment (dilallkx) [inline js-apps/meep-frontend/src/js/util/elem-utils.js:261]: Remove this.

File: js-apps/meep-frontend/src/js/util/elem-utils.js [hunk 4]
```diff
@@ -130,6 +134,8 @@ export const FIELD_APP_THROUGHPUT_UL = 'appThroughput_Ul';
 export const FIELD_APP_PKT_LOSS = 'appPacketLoss';
 export const FIELD_META_DISPLAY_MAP_COLOR = 'metaDisplayMapColor';
 export const FIELD_META_DISPLAY_MAP_ICON = 'metaDisplayMapIcon';
+export const FIELD_META_DISPLAY_SEQ_PARTICIPANTS = 'metaDisplaySeqParticipants';
```
Comment (dilallkx) [inline js-apps/meep-frontend/src/js/util/elem-utils.js:137]: Remove this field.

**Review by dilallkx — CHANGES_REQUESTED** [review]

File: js-apps/meep-frontend/src/js/util/elem-utils.js (hunk 3 above)
Comment (dhruvigoyal25) [reply js-apps/meep-frontend/src/js/util/elem-utils.js]: Removed.

File: js-apps/meep-frontend/src/js/util/elem-utils.js (hunk 4 above)
Comment (dhruvigoyal25) [reply js-apps/meep-frontend/src/js/util/elem-utils.js]: Removed.

**Pull request merged**